sklearn/manifold/_isomap.py

```diff
@@ -294,7 +294,14 @@ def _fit_transform(self, X):
 
         self.dist_matrix_ = shortest_path(nbg, method=self.path_method, directed=False)
 
+        if np.array(X).dtype == np.float32:
+            self.dist_matrix_ = self.dist_matrix_.astype(X.dtype, copy=False)
```

Comment:
With `copy=False`, you can remove this test as only `np.float{32,64}` are accepted.
```suggestion
      self.dist_matrix_ = self.dist_matrix_.astype(X.dtype, copy=False)
```

Reply:
(Ideally, `scipy.sparse.shortest_path` could support float32, I need to look into it.)

Reply:
I included all the changes except this one. `pytest sklearn/tests/test_common.py -v -k Isomap` fails 2 tests if I make this change.